sriov/sriov.go

```diff
@@ -49,36 +51,31 @@ func loadConf(bytes []byte, args string) (*SriovConf, error) {
 }
 
 func setupVF(conf *SriovConf, ifName string, netns ns.NetNS) error {
+	var err error
 	vfIdx := 0
 	masterName := conf.Net.Master
 	args := conf.Args
 
 	if args.VF != 0 {
 		vfIdx = args.VF
+	} else {
+		// alloc a free virtual function
+		if vfIdx, err = allocFreeVF(masterName); err != nil {
+			return err
+		}
 	}
-	// TODO: if conf.VF == nil, alloc vf randomly
 
 	m, err := netlink.LinkByName(masterName)
 	if err != nil {
 		return fmt.Errorf("failed to lookup master %q: %v", masterName, err)
 	}
 
-	vfDir := fmt.Sprintf("/sys/class/net/%s/device/virtfn%d/net", masterName, vfIdx)
-	if _, err := os.Lstat(vfDir); err != nil {
-		return err
-	}
-
-	infos, err := ioutil.ReadDir(vfDir)
+	// VF device name
+	vfDevName, err := getVFDeviceName(masterName, vfIdx)
```

Comment:
Do we need to call the getVFDeviceName() twice inside the allocFreeVF() function ?